tests/components/sonos/test_switch.py

```diff
@@ -45,3 +46,31 @@ async def test_alarm_attributes(hass, config_entry, config, soco):
     assert alarm_state.attributes.get(ATTR_VOLUME) == 0.25
     assert alarm_state.attributes.get(ATTR_PLAY_MODE) == "SHUFFLE_NOREPEAT"
     assert not alarm_state.attributes.get(ATTR_INCLUDE_LINKED_ZONES)
+
+
+async def test_alarm_create_delete(
+    hass, config_entry, config, soco, alarm_clock, alarm_clock_extended, alarm_event
+):
+    """Test for correct creation and deletion of alarms during runtime."""
+    await setup_platform(hass, config_entry, config)
+
+    soco.alarmClock = alarm_clock_extended
+    speaker = hass.data[DATA_SONOS].discovered[soco.uid]
```

Comment:
Please don't access integration details in the tests. Patch the library to get access to callbacks if needed. Then trigger the callback and assert some relevant core state.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
While I do agree 100%, that is not how it has been done in the other tests of the sonos component.
See e.g.: https://github.com/home-assistant/core/blob/cede36d91c6dce4109791cce2de30a5f45830f08/tests/components/sonos/test_media_player.py#L23

I will leave it like that and maybe someone will refactor all these occurrences in the future.